public/app/controllers/mainCtrl.js

```diff
@@ -40,63 +37,25 @@ angular.module('kibibitCodeEditor')
         controller: 'projectFolderModalController',
         controllerAs: 'projectFolderModalCtrl',
         className: 'ngdialog-theme-default',
-        scope: $scope
+        scope: $scope,
+        data: {
+          userHomeDirectoryPath: vm.userHomeDirectoryPath
+        }
+      }).closePromise.then(function(selectedProjectFolder) {
+        EventManagerService.trigger(
```

Comment:
If we're updating the mainCtrl we'll have to put a watcher in fileTree link function
(in order to trigger folder content loading)
should I do it?